lib/matplotlib/legend.py

```diff
@@ -1141,27 +1149,19 @@ def _get_legend_handles(axs, legend_handler_map=None):
     for ax in axs:
         handles_original += [
             *(a for a in ax._children
-              if isinstance(a, (Line2D, Patch, Collection))),
+              if isinstance(a, (Line2D, Patch, Collection, Text))),
             *ax.containers]
         # support parasite axes:
         if hasattr(ax, 'parasites'):
             for axx in ax.parasites:
                 handles_original += [
                     *(a for a in axx._children
-                      if isinstance(a, (Line2D, Patch, Collection))),
+                      if isinstance(a, (Line2D, Patch, Collection, Text))),
                     *axx.containers]
 
-    handler_map = Legend.get_default_handler_map()
```

Comment:
Why did you remove this whole code block?  Without this here one of the inputs in now completely and silently ignored.

Reply:
Hm. I don't remember the call sequence well enough to know why I did it in the first place. But I have an example where I pass a `handler_map` dict in the `legend` call and this block doesn't update the handler map:
```python
ax.legend(handles, labels, handler_map={Text: HandlerText()})
```
so I might've deleted it because it did nothing.

Reply:
That may be evidence that there is a bug someplace else that is making it so the additional handler map is not getting routed.

I suspect that the change we want to make in this function is to drop the filtering on L1144 all together, put this block of code back, filter out the things with `_nolegend_` as the label, warn on things that do not have a handler (but drop them on the floor), and then restore the behavior of only yielding things that have handlers (that way we can give a different warning for auto-discovered things rather than explicitly passed things).

If we do this then I do not think we need any changes to `_init_legend_box` ?